discovery-provider/src/api/v1/users.py

```diff
@@ -743,6 +750,54 @@ def get(self, id):
         return success_response(favorites)
 
 
+USER_TRACKS_LIBRARY_ROUTE = "/<string:id>/library/tracks"
+
+
+@full_ns.route(USER_TRACKS_LIBRARY_ROUTE)
+class UserTracksLibraryFull(Resource):
+    @record_metrics
+    @full_ns.doc(
+        id="Get User Library Tracks",
+        description="Gets a user's saved/reposted/purchased/all tracks",
+        params={"id": "A user ID"},
+        responses={200: "Success", 400: "Bad request", 500: "Server error"},
+    )
+    @full_ns.expect(user_tracks_library_parser)
+    @full_ns.marshal_with(favorites_full_response)
+    @auth_middleware()
+    @cache(ttl_sec=5)
+    def get(self, id: str, authed_user_id: Optional[int] = None):
+        """Fetch a user's full library tracks."""
+        args = user_tracks_library_parser.parse_args()
+        decoded_id = decode_with_abort(id, ns)
+        current_user_id = get_current_user_id(args)
+        if authed_user_id != current_user_id and authed_user_id != id:
```

Comment:
nit: Do we need to bother with if `current_user_id` isn't specified?

Reply:
Very good point... I don't think we even need current_user_id, since this is authed AND we have user_id already. Going to remove this, redundant arg

Reply:
Need to keep this around in the query for backwards compat but removing from the API layer